shared/config/config.go

```diff
@@ -531,23 +516,134 @@ func (r configReader) GetUSDCDenom(chainID string) (string, error) {
 		return "", fmt.Errorf("chain id %s not found", chainID)
 	}
 
-	switch chainConfig.Type {
+	return chainConfig.USDCDenom, nil
+}
+
+func ValidateChainConfig(chain ChainConfig) error {
+	if chain.ChainName == "" {
+		return fmt.Errorf("chain_name is required")
+	}
+	if chain.ChainID == "" {
+		return fmt.Errorf("chain_id is required")
+	}
+	if chain.Type == "" {
+		return fmt.Errorf("type is required")
+	}
+	if chain.Environment == "" {
+		return fmt.Errorf("environment is required")
+	}
+	if chain.GasTokenSymbol == "" {
+		return fmt.Errorf("gas_token_symbol is required")
+	}
+	if chain.GasTokenDecimals == 0 {
+		return fmt.Errorf("gas_token_decimals is required")
+	}
+	if chain.NumBlockConfirmationsBeforeFill == 0 {
+		return fmt.Errorf("num_block_confirmations_before_fill is required")
+	}
+	if chain.HyperlaneDomain == "" {
+		return fmt.Errorf("hyperlane_domain is required")
+	}
+	if chain.QuickStartNumBlocksBack == 0 {
+		return fmt.Errorf("quick_start_num_blocks_back is required")
+	}
+	if chain.MaxRebalancingGasThreshold == 0 {
+		return fmt.Errorf("max_rebalancing_gas_threshold is required")
+	}
+	if chain.FastTransferContractAddress == "" {
+		return fmt.Errorf("fast_transfer_contract_address is required")
+	}
+	if chain.SolverAddress == "" {
+		return fmt.Errorf("solver_address is required")
+	}
+	if chain.USDCDenom == "" {
+		return fmt.Errorf("usdc_denom is required")
+	}
+	if chain.BatchUUSDCSettleUpThreshold == "" {
+		return fmt.Errorf("batch_uusdc_settle_up_threshold is required")
+	}
+	if chain.MinFeeBps == 0 {
+		return fmt.Errorf("min_fee_bps is required")
+	}
+	if chain.MinProfitMarginBPS == 0 {
+		return fmt.Errorf("min_profit_margin_bps is required")
+	}
+	if chain.MinProfitMarginBPS >= chain.MinFeeBps {
```

Comment:
ah sorry should have specified the should just be > not >=
```suggestion
	if chain.MinProfitMarginBPS > chain.MinFeeBps {
```